v2/pkg/operators/common/dsl/dsl.go

```diff
@@ -447,6 +448,26 @@ func init() {
 				return true, nil
 			},
 		),
+		"compare_versions": makeDslWithOptionalArgsFunction(
```

Comment:
Shouldn't you rather use the `makeDslFunction` instead? The parameters for this method should not be optional.

Reply:
Also wouldn't be easier to compare two versions and return `-1, 0, 1`, similarly to the `strings.Compare` function?

Reply:
We use go-version from hashicorp, which supports a standard format for parsing version constraints.

https://github.com/hashicorp/go-version/blob/main/constraint.go#L40-L48 -> These are the operators. The spaces are ignored. Regarding multiple values, Imo it makes more sense in a single param for all conditions.

Reply:
> Imo it makes more sense in a single param for all conditions since it will look more uniform

What do you mean by more uniform? We already have function(s) that accept varargs. As you know I am not a fan of working with strings (parsing user input where it's not strictly necessary), and I prefer stricter approaches, where fewer things can go wrong.

That being said (even though I don't like it, because it's not a good practice, nor is it straight forward from a UX point of view), I could live with it, because we are basically just delegating the parameters to a 3rd party library. Still, I would prefer the vararg approach :P

Reply:
Done